File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var twoSumLessThanK = function(nums, k) {
    nums.sort(function(a,b) { return a - b })
    let i = 0, j = nums.length-1, ans = -1
    while(i < j) {
        let x = nums[i], y = nums[j]
        if(x+y < k) {
            ans = Math.max(ans,x+y)
            i++
        } else {
            j--
        }
    }
    return ans
};

File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.java
class Solution {
    public int twoSumLessThanK(int[] nums, int k) {
        int n = nums.length;
        int ans = Integer.MIN_VALUE;
        for(int i = 0; i < n; i++) {
            for(int j = i + 1; j < n; j++) {
                if(nums[i] + nums[j] < k) {
                    ans = Math.max(ans, nums[i] + nums[j]);
                }
            }
        }
        if(ans == Integer.MIN_VALUE) return -1;
        return ans;
    }
}